### ISSUE TITLE
Hyperparameters of  Fan model on Logical inference task 

### ISSUE THREAD

**Issue by Demon-JieHao** [ketranm/fan_vs_rnn#1]
Hi, I am trying to reproduce the Fan model result on logical inference. But the results seems far away from the original results in the paper. What are the hyperparameters of the Fan model in this task?  Thanks!

**Comment by ketranm** [comment]
Hi, I ran a grid search over hyperparameters. Number of layers {2, 3, 4}, dropout {0.2, 0.3, 0.5}, learning rate {0.00001, 0.0001, 0.001}, # hidden units {138, 256, 512} and #heads {2, 4}. The details are in our paper http://aclweb.org/anthology/D18-1503 under *hyperparameters*. Unfortunately I don't remember the best configuration on top of my head.

**Comment by Demon-JieHao** [comment]
Thanks for answering my question. When I train the Fan model on logical inference task, the eval score are always same after each epoch, even in different hyper-parameter configurations. If I change it to Rnn, the training process and results return to normal.  It is confusing for me.

**Comment by hsing-wang** [comment]
Hi, my result of Fan model on logical infer task is quite low as well under grid search over hyper-parameters. What are the epochs, batch_size and param_init in your setting? Thank you very much.

**Comment by hsing-wang** [comment]
One of my results are here. 
![143144444](https://user-images.githubusercontent.com/3795236/52042219-92752f00-2577-11e9-88c0-52b6ca8cf964.PNG)

**Comment by ketranm** [comment]
Hi there, the hyperparameters are provided in our paper. Parameters are initialized uniformly in (0, 0,1). The batch size is set to 64 (I think). All the models are trained for maximum 50 epochs.

The results of FAN models are usually unstable and very sensitive to hyperparameters unless some heuristic learning rate decay method is used.

Also, which PyTorch version are you using? There might be a bug if you use PyTorch 1.0.

**Comment by hsing-wang** [comment]
Hello, thanks! I am using PyTorch 0.4.1. There's a bug in the "add_timing_signal" function. log_inv_inc should be converted to float. 
![default](https://user-images.githubusercontent.com/3795236/52049095-d96b2080-2587-11e9-805a-aa0b23ce7bbf.PNG)

By the way, are you using Skorch to conduct the grid search in Pytorch?  I don't know how to do grid search in a right way in PyTorch.
